pkg/cmd/issue/comment/comment.go

```diff
@@ -0,0 +1,199 @@
+		Example: heredoc.Doc(`
+			$ gh issue comment 22 --body "I was able to reproduce this issue, lets fix it."
+		`),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+			opts.SelectorArg = args[0]
+
+			inputFlags := 0
+			if cmd.Flags().Changed("body") {
+				opts.InputType = inline
+				inputFlags++
+			}
+			if webMode {
+				opts.InputType = web
+				inputFlags++
+			}
+			if editorMode {
+				opts.InputType = editor
+				inputFlags++
+			}
+
+			if inputFlags == 0 {
+				if !opts.IO.CanPrompt() {
+					return &cmdutil.FlagError{Err: errors.New("--body or --web required when not running interactively")}
+				}
+				opts.Interactive = true
+			} else if inputFlags == 1 {
+				if !opts.IO.CanPrompt() && opts.InputType == editor {
+					return &cmdutil.FlagError{Err: errors.New("--body or --web required when not running interactively")}
+				}
+			} else if inputFlags > 1 {
+				return &cmdutil.FlagError{Err: fmt.Errorf("specify only one of --body, --editor, or --web")}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return commentRun(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.Body, "body", "b", "", "Supply a body. Will prompt for one otherwise.")
+	cmd.Flags().BoolVarP(&editorMode, "editor", "e", false, "Add body using editor")
+	cmd.Flags().BoolVarP(&webMode, "web", "w", false, "Add body in browser")
+
+	return cmd
+}
+
+func commentRun(opts *CommentOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+	apiClient := api.NewClientFromHTTP(httpClient)
+
+	issue, baseRepo, err := shared.IssueFromArg(apiClient, opts.BaseRepo, opts.SelectorArg)
+	if err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		inputType, err := opts.InputTypeSurvey()
+		if err != nil {
+			return err
+		}
+		opts.InputType = inputType
+	}
+
+	switch opts.InputType {
+	case web:
+		openURL := issue.URL + "#issuecomment-new"
+		if opts.IO.IsStdoutTTY() {
+			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(openURL))
+		}
+		return opts.OpenInBrowser(openURL)
+	case editor:
+		body, err := opts.EditSurvey()
+		if err != nil {
+			return err
+		}
+		opts.Body = body
+		if opts.Interactive {
+			cs := opts.IO.ColorScheme()
+			fmt.Fprint(opts.IO.Out, cs.GreenBold("?"))
+			fmt.Fprint(opts.IO.Out, cs.Bold(" Body "))
+			fmt.Fprint(opts.IO.Out, cs.Cyan("<Received>\n"))
+		}
+	}
+
+	if opts.Interactive {
+		cont, err := opts.ConfirmSubmitSurvey()
+		if err != nil {
+			return err
+		}
+		if !cont {
+			return fmt.Errorf("Discarding...")
+		}
+	}
+
+	params := map[string]string{
+		"subjectId": issue.ID,
+		"body":      opts.Body,
+	}
+	url, err := api.CommentCreate(apiClient, baseRepo.RepoHost(), params)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintln(opts.IO.Out, url)
+	return nil
+}
+
+var inputTypeSurvey = func() (int, error) {
+	var inputType int
```

Comment:
This value is `0` when "Editor" was selected and `1` if "Web" was selected. This just so happens to match the values of `iota`-generated `editor` and `web` consts, since they start from 0. However, if anyone adds new iota constants above the `editor..web` block, that would shift the values of these constants and they would no longer match Survey results. So I don't feel it's safe to count on this behavior.

Instead, I would explicitly translate Survey results into input mode constants:
```go
switch surveyResult {
case 0: return inputTypeEditor
case 1: return inputTypeWeb
}
```
Additionally, I think it's good practice to define a new type for these constants, e.g.
```
type inputType int
```
And then change this function to return `inputType` instead of int. That way the reader is clear what kind of values are being passed around, instead of having to guess what `int` represents.